pkg/pillar/agentlog/agentlog.go

```diff
@@ -28,23 +30,37 @@ var savedAgentName = "unknown" // Keep for signal and exit handlers
 var savedRebootReason = "unknown"
 var savedPid = 0
 
+func setupSyslog(agentName string) error {
+	log.SetOutput(ioutil.Discard)
+	syslogPriorities := syslog.LOG_INFO | syslog.LOG_DEBUG |
+		syslog.LOG_ERR | syslog.LOG_WARNING | syslog.LOG_NOTICE |
+		syslog.LOG_CRIT | syslog.LOG_ALERT | syslog.LOG_EMERG
+
+	syslogHook, err := lSyslog.NewSyslogHook("udp", "127.0.0.1:514",
+		syslogPriorities, agentName)
+	if err != nil {
+		fmt.Printf("setupSyslog: Failed creating syslog hook with error: %s", err)
+		return err
+	}
+	log.AddHook(syslogHook)
+	return nil
+}
+
 // Parameter description
 // 1. agentName: Name with which disk log file will be created.
 // 2. logdir: Directory in which disk log file will be placed.
 // 3. text: Test based loggin i.e. logs do into a file created based on
 //          the argument values passed for agentName & logdir.
 //          examples: logmanager
-func initImpl(agentName string, logdir string, text bool) (*os.File, error) {
```

Comment:
Furthermore "text" is never true so you can remove more code, right? And the above sysloghook code.

Reply:
removed.